test/middleware.test.js

```diff
@@ -513,6 +514,45 @@ test('middlewares should support encapsulation with prefix', t => {
   })
 })
 
+test('middlewares should support plugins', t => {
```

Comment:
this should be `'middlewares should support not-encapsulated plugins'`. This tests checks plugins defined with  `fastify-plugin`, i.e. the ones that are not encapsulated.